contracts/CarbonVoteXCore.sol

```diff
@@ -0,0 +1,254 @@
+pragma solidity ^0.4.23;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./ICarbonVoteXReceiver.sol";
+import "./guard.sol";
+
+
+contract CarbonVoteXCore is DSGuard {
+    // apply SafeMath to uint
+    using SafeMath for uint;
+
+    // events that are emitted after changes.
+    event _Register(
+        address indexed msgSender,
+        bytes32 namespace,
+        bytes32 pollId,
+        uint startBlock,
+        uint endBlock,
+        address tokenAddr
+    );
+    event _WriteAvailableVotes (
+        address indexed msgSender,
+        bytes32 namespace,
+        bytes32 pollId, 
+        address voter, 
+        uint votes
+    );
+    event _SendGas(
+        address indexed msgSender, 
+        bytes32 namespace, 
+        bytes32 pollId,
+        uint value
+    );
+    
+    struct Poll {
+        // map if a voter has already obtained voting right
+        mapping (address => bool) obtained;
+        // map voter's address to the amount of gas sent
+        mapping (address => uint) gasSentByVoter;
+        uint startBlock;
+        uint endBlock;
+        bytes32 pollId;
+        // the ERC20 token used
+        ERC20 token;
+    }
+
+    // namespace => pollId => Poll struct
+    mapping(bytes32 => mapping (bytes32 => Poll)) public polls;
+
+    // namespace => receiver addresses
+    mapping(bytes32 => address) receiverAddr;
+
+    // master's address
+    address private master;
+
+    // map the hashed function name to the signature of the function
+    mapping(bytes32 => bytes4) public functionSig;
+
+    /*
+     * Constructor
+     */
+    // @param master address which is able to write votes,
```

Comment:
the address, which is able to write votes

or 
the address that is able to write votes

Reply:
Actually, root account can also call writeAvailableVotes
should use " the address, which is able to write votes"